main.py

```diff
@@ -152,29 +164,42 @@ def get_response_data():
         print(e)
         return jsonify({"error": "".format(e)}), 400
 
+    # Assuming you have the correct neo4j_driver instance.
+    
     try:
         if neo4j_driver:
+            print(response_data)
             # Import nodes
-            neo4j_driver.execute_query("""
-            UNWIND $nodes AS node
-            MERGE (n:Node {id:toLower(node.id)})
-            SET n.type = node.type, n.label = node.label, n.color = node.color""",
-                                       {"nodes": response_data['nodes']})
+            results = neo4j_driver.execute_query(
+                """
+                UNWIND $nodes AS node
+                MERGE (n:Node {id: node.id})
+                SET n.type = node.type, n.label = node.label, n.color = node.color
+                """,
+                {"nodes": response_data['nodes']}
+            )
+            print("Results from Neo4j:", results)
+            #print(f"Created {summary.counters.updates().nodesCreated} nodes.")
             # Import relationships
-            neo4j_driver.execute_query("""
-            UNWIND $rels AS rel
-            MATCH (s:Node {id: toLower(rel.from)})
-            MATCH (t:Node {id: toLower(rel.to)})
-            MERGE (s)-[r:RELATIONSHIP {type:rel.relationship}]->(t)
-            SET r.direction = rel.direction,
-
-                r.color = rel.color,
-                r.timestamp = timestamp();
-            """, {"rels": response_data['edges']})
-
-    except json.decoder.JSONDecodeError as jde:
-        return jsonify({"Error": "{}".format(jde)}), 500
-
+            results = neo4j_driver.execute_query(
+                """
+                UNWIND $rels AS rel
+                MATCH (s:Node {id: rel.from})
+                MATCH (t:Node {id: rel.to})
+                MERGE (s)-[r:RELATIONSHIP {type:rel.relationship}]->(t)
+                SET r.direction = rel.direction,
+                    r.color = rel.color,
+                    r.timestamp = timestamp();
+                """,
+                {"rels": response_data['edges']}
+            )
+            #print(f"Created {summary.counters.updates().relationshipsCreated} relationships.")
+            print("Results from Neo4j:", results)
+            
```

Comment:
if You add the following bit of code after graphQL insertion. You can create a graph without doing the second API call from front-end. Thus saving some ~(5sec-15sec) from front-end -> backend -> database -> backend -> front-end.

```python
# create a payload to return.
            nodes = [
                {
                    "data": {
                        "id": node["id"],
                        "label": node["label"],
                        "color": node.get("color", "defaultColor"),
                    }
                }
                for node in response_data["nodes"]
            ]

            edges = [
                {
                    "data": {
                        "source": edge["from"],
                        "target": edge["to"],
                        "label": edge["relationship"],
                        "color": edge.get("color", "defaultColor"),
                        "direction": edge["direction"],
                    }
                }
                for edge in response_data["edges"]
            ]

            return jsonify({
                "elements": {
                    "nodes": nodes,
                    "edges": edges
                },}
                
```

I will mark the related changes in `index.html` as well.